src/dotnet/APIView/apiview.yml

```diff
@@ -30,230 +38,367 @@ pr:
+                          echo Installing npm packages
+                          call npm install --prefix D:\home\site\wwwroot --force
+                          call npm run-script build --prefix D:\home\site\wwwroot --force
+                          call D:\home\site\wwwroot\Python\python -m pip uninstall api-stub-generator
+                          call D:\home\site\wwwroot\Python\python -m pip install pylint==2.13.9 pylint-guidelines-checker==0.0.6 api-stub-generator==0.3.3 --index-url "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/"
+                        enableCustomDeployment: true
+                        DeploymentType: zipDeploy
+
+
+          - ${{ if eq(parameters.Version, 'staging') }}:
+            - deployment: Publish_to_Staging
+              displayName: Publish to Staging
+              environment: 'package-publish'
+              pool:
+                name: azsdk-pool-mms-ubuntu-2004-general
+                image: azsdk-pool-mms-ubuntu-2004-1espt
+                os: linux
+
+              strategy:
+                runOnce:
+                  deploy:
+                    steps:
+                    - checkout: self
+
+                    - download: current
+                      artifact: APIView
+                      timeoutInMinutes: 5
+
+                    - task: FileTransform@1
+                      displayName: 'Update Angular Config for Environment'
+                      inputs:
+                        folderPath: '$(Pipeline.Workspace)/APIView/**/APIViewWeb.zip'
+                        fileType: json
+                        targetFiles: wwwroot/spa/assets/config.json
+
+                    - task: AzureRmWebAppDeployment@4
+                      displayName: 'Deploy Azure App Service'
+                      inputs:
+                        azureSubscription: 'APIview staging resource group'
+                        appType: 'webApp'
+                        WebAppName: 'apiviewstaging'
+                        packageForLinux: '$(Pipeline.Workspace)/APIView/**/APIViewWeb.zip'
+                        ScriptType: 'Inline Script'
+                        InlineScript: |
+                          @echo off
+                          echo Installing npm packages
+                          call npm install --prefix D:\home\site\wwwroot --force
+                          call npm run-script build --prefix D:\home\site\wwwroot --force
+                          call D:\home\site\wwwroot\Python\python -m pip uninstall api-stub-generator
+                          call D:\home\site\wwwroot\Python\python -m pip install pylint==2.13.9 pylint-guidelines-checker==0.0.6 api-stub-generator==0.3.2 --index-url "https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/"
+                        enableCustomDeployment: true
+                        DeploymentType: zipDeploy
+
+          - ${{ if eq(parameters.Version, 'staging') }}:
```

Comment:
Can we use different param value to enable this stage? for e.g. `test`. We don't want all the deployments that goes to staging goes here in Test UI instance.

Reply:
I'm not doing exactly this, but a version of it. Updates incoming.

Reply:
We have done this. The Test UI Instance will show up _alongside_ the automatic staging deployment as a separate approvable job.